src/main/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessor.java

```diff
@@ -306,9 +310,42 @@ private <T extends SchemaObject> Tags getCustomTags(
     JsonApiMetricsConfig.SortType sortType = getVectorTypeTag(commandContext, command);
     Tag sortTypeTag = Tag.of(jsonApiMetricsConfig.sortType(), sortType.name());
 
+    // --- Command Feature Usage Tags ---
+    if (command instanceof FeatureSource fs) {
+      commandContext.addFeatureUsage(fs.getFeatureUsage());
+    }
+    Tags commandFeatureTags = getCommandFeatureTags(commandContext.featureUsage());
+
     // --- Combine All Tags ---
     return Tags.of(
-        commandTag, tenantTag, errorTag, errorClassTag, errorCodeTag, vectorEnabled, sortTypeTag);
+            commandTag,
+            tenantTag,
+            errorTag,
+            errorClassTag,
+            errorCodeTag,
+            vectorEnabled,
+            sortTypeTag)
+        .and(commandFeatureTags);
+  }
+
+  /**
+   * Adds tags for all defined features. If a feature is present in FeatureUsage, its tag value is
+   * "true", otherwise "false".
+   */
+  private Tags getCommandFeatureTags(FeatureUsage featureUsage) {
```

Comment:
This shoudl be on the FeatureUsage class, that is the holder of all of the features